packages/time-picker/src/vaadin-time-picker.js

```diff
@@ -627,10 +628,24 @@ class TimePicker extends PatternMixin(InputControlMixin(ThemableMixin(ElementMix
     }
   }
 
+  /** @private */
+  __onComboBoxCustomValueSet(event) {
+    const value = event.detail;
+    const parsedObj = this.i18n.parseTime(value);
```

Comment:
It seems that it's not required to split the logic between `custom-value-set` and `change` handlers, as everything can be done in the `__onComboBoxChange` listener.

Reply:
The idea behind this listener is to only parse custom values (so there is no need to call `parseTime` when the value is present in the dropdown list). But yes, we can simplify this indeed.